src/main/java/book/store/service/user/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package book.store.service.user.impl;
+
+import book.store.dto.user.UserRegistrationRequestDto;
+import book.store.dto.user.UserResponseDto;
+import book.store.exception.RegistrationException;
+import book.store.mapper.UserMapper;
+import book.store.model.User;
+import book.store.repository.user.UserRepository;
+import book.store.service.user.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+class UserServiceImpl implements UserService {
+    private static final String USER_ALREADY_EXIST = "User already exist!";
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findUserByEmail(requestDto.getEmail()).isPresent()) {
```

Comment:
```suggestion
        if (userRepository.existsByEmail(requestDto.getEmail())) {
```
create `boolean existsByEmail(String email);` to avoid calling isPresent()